src/agent/custom-http-agent.ts

```diff
@@ -55,9 +62,14 @@ export class CustomHttpAgent {
     arg,
     canisterId,
     method: methodName,
-    nonce
-  }: Omit<IcrcCallCanisterRequestParams, 'sender'>): Promise<CustomHttpAgentResponse> => {
+    nonce,
+    sender
+  }: IcrcCallCanisterRequestParams): Promise<CustomHttpAgentResponse> => {
+    const hash = await generateHash({canisterId, sender, method: methodName, arg, nonce});    
```

Comment:
If there is no nonce provided in the request we probably do not need to use the cache and can solely attach a transform expiry with default duration.